src/drivers/ist8310/ist8310.cpp

```diff
@@ -985,14 +939,30 @@ IST8310::collect()
 	report.y = (((int16_t)report_buffer.y[1]) << 8) | (int16_t)report_buffer.y[0];
 	report.z = (((int16_t)report_buffer.z[1]) << 8) | (int16_t)report_buffer.z[0];
 
-	/* perform cross-axis compensation */
-//	crossaxis_transformation((int16_t *)&report);
+
+	/* FSR:
+	 *   x, y: +- 1600 µT
+	 *   z:    +- 2500 µT
+	 *
+	 * Internal ADC is 14 bits, so also reduce z to +- 2457.3 µT.
+	 * Check if value makes sense, and discard any outlier
+	 */
+	if (report.x > 5334 || report.x < -5334 ||
+	    report.y > 5334 || report.y < -5334 ||
```

Comment:
It's basically checking for values within the FSR as per comment above, but using sensor units instead of converting to µT:

FSR for x, y is 1600 µT
Resolution is 0.3 µT/LSB
FSR in LSB units is 1600/0.3 == 5333.3.